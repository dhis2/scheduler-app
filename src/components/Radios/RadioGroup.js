import React from 'react'
import { arrayOf, shape, func, string } from 'prop-types'
import { Radio } from '@dhis2/ui-core'

const RadioGroup = ({ name, options, selected, setSelected }) => {
    return (
        <React.Fragment>
            {options.map(({ value, label }) => (
                <Radio
                    key={value}
                    label={label}
                    name={name}
                    onChange={event => setSelected(event.target.value)}
                    value={value}
                    checked={selected === value}
                />
            ))}
        </React.Fragment>
    )
}

RadioGroup.propTypes = {
    name: string.isRequired,
    options: arrayOf(
        shape({
            value: string.isRequired,
            label: string.isRequired,
        })
    ).isRequired,
    selected: string.isRequired,
    setSelected: func.isRequired,
}

export default RadioGroup
