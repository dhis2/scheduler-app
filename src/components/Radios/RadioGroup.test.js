import React from 'react'
import { shallow, mount } from 'enzyme'
import RadioGroup from './RadioGroup'

describe('<RadioGroup>', () => {
    it('renders correctly', () => {
        const props = {
            options: [
                {
                    value: 'one',
                    label: 'one',
                },
                {
                    value: 'two',
                    label: 'two',
                },
            ],
            name: 'name',
            selected: '',
            setSelected: () => {},
        }
        const wrapper = shallow(<RadioGroup {...props} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('calls setSelected when a radio button is clicked', () => {
        const props = {
            options: [
                {
                    value: 'one',
                    label: 'one',
                },
            ],
            name: 'name',
            selected: '',
            setSelected: jest.fn(),
        }
        const wrapper = mount(<RadioGroup {...props} />)

        wrapper
            .find('input')
            .simulate('change', { target: { value: 'change' } })

        expect(props.setSelected).toHaveBeenCalledWith('change')
    })

    it('checks the radio button if its value matches selected', () => {
        const props = {
            options: [
                {
                    value: 'one',
                    label: 'one',
                },
            ],
            name: 'name',
            selected: 'one',
            setSelected: () => {},
        }
        const wrapper = mount(<RadioGroup {...props} />)

        const input = wrapper.find('input')

        expect(input.props().checked).toEqual(true)
    })
})
