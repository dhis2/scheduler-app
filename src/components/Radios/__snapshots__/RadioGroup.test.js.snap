// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup> renders correctly 1`] = `
<Fragment>
  <Radio
    checked={false}
    key="one"
    label="one"
    name="name"
    onChange={[Function]}
    value="one"
  />
  <Radio
    checked={false}
    key="two"
    label="two"
    name="name"
    onChange={[Function]}
    value="two"
  />
</Fragment>
`;
